fix automatic tests after w3emc pr

**Describe the bug**
After PR https://github.com/NOAA-EMC/WW3/pull/745 the automatic tests need to be updated with the module updates as the automatic tests failed. 


**Expected behavior**
Automatic tests should pass.

Although w3nco is changed to w3emc in [spack.yaml](https://github.com/NOAA-EMC/WW3/blob/develop/model/ci/spack.yaml), cmake still searches for w3nco in github action:


```
CMake Error at model/src/CMakeLists.txt:188 (find_package):
  By not providing "Findw3nco.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "w3nco", but
  CMake did not find one.

  Could not find a package configuration file provided by "w3nco" with any of
  the following names:

    w3ncoConfig.cmake
    w3nco-config.cmake

  Add the installation prefix of "w3nco" to CMAKE_PREFIX_PATH or set
  "w3nco_DIR" to a directory containing one of the above files.  If "w3nco"
  provides a separate development package or SDK, be sure it has been
  installed.
```

--- a/.github/workflows/gnu.yml
+++ b/.github/workflows/gnu.yml
@@ -1,11 +1,17 @@
 name: GNU Linux Build
 on: [push, pull_request]
 
+# Cancel in-progress workflows when pushing to a branch
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
 env:
-  cache_key: gnu4
+  cache_key: gnu5
   CC: gcc-10
   FC: gfortran-10
   CXX: g++-10
+
 
 # Split into a steup step, and a WW3 build step which
 # builds multiple switches in a matrix. The setup is run once and 
@@ -16,6 +22,11 @@ jobs:
     runs-on: ubuntu-20.04
 
     steps:
+      - name: checkout-ww3
+        if: steps.cache-env.outputs.cache-hit != 'true'
+        uses: actions/checkout@v2
+        with: 
+            path: ww3
       # Cache spack, OASIS, and compiler
       # No way to flush Action cache, so key may have # appended
       - name: cache-env
@@ -26,13 +37,7 @@ jobs:
             spack
             ~/.spack
             work_oasis3-mct
-          key: spack-${{ runner.os }}-${{ env.cache_key }}
-
-      - name: checkout-ww3
-        if: steps.cache-env.outputs.cache-hit != 'true'
-        uses: actions/checkout@v2
-        with: 
-            path: ww3
+          key: spack-${{ runner.os }}-${{ env.cache_key }}-${{ hashFiles('ww3/model/ci/spack.yaml') }}
 
       # Build WW3 spack environment
       - name: install-dependencies-with-spack
@@ -83,7 +88,7 @@ jobs:
             spack
             ~/.spack
             work_oasis3-mct
-          key: spack-${{ runner.os }}-${{ env.cache_key }}
+          key: spack-${{ runner.os }}-${{ env.cache_key }}-${{ hashFiles('ww3/model/ci/spack.yaml') }}
 
       - name: build-ww3
         run: |

--- a/.github/workflows/intel.yml
+++ b/.github/workflows/intel.yml
@@ -1,6 +1,11 @@
 name: Intel Linux Build
 on: [push, pull_request]
 
+# Cancel in-progress workflows when pushing to a branch
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
 # Use custom shell with -l so .bash_profile is sourced which loads intel/oneapi/setvars.sh
 # without having to do it in manually every step
 defaults:
@@ -9,7 +14,7 @@ defaults:
 
 # Set I_MPI_CC/F90 so Intel MPI wrapper uses icc/ifort instead of gcc/gfortran
 env:
-  cache_key: intel5
+  cache_key: intel6
   CC: icc
   FC: ifort
   CXX: icpc
@@ -25,6 +30,13 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
+
+      - name: checkout-ww3
+        if: steps.cache-env.outputs.cache-hit != 'true'
+        uses: actions/checkout@v2
+        with: 
+            path: ww3
+
       # Cache spack, OASIS, and compiler
       # No way to flush Action cache, so key may have # appended
       - name: cache-env
@@ -36,7 +48,7 @@ jobs:
             ~/.spack
             work_oasis3-mct
             /opt/intel
-          key: spack-${{ runner.os }}-${{ env.cache_key }}
+          key: spack-${{ runner.os }}-${{ env.cache_key }}-${{ hashFiles('ww3/model/ci/spack.yaml') }}
 
       - name: install-intel-compilers
         if: steps.cache-env.outputs.cache-hit != 'true'
@@ -47,12 +59,6 @@ jobs:
           sudo apt-get update
           sudo apt-get install intel-oneapi-dev-utilities intel-oneapi-mpi-devel intel-oneapi-openmp intel-oneapi-compiler-fortran intel-oneapi-compiler-dpcpp-cpp-and-cpp-classic
           echo "source /opt/intel/oneapi/setvars.sh" >> ~/.bash_profile
-
-      - name: checkout-ww3
-        if: steps.cache-env.outputs.cache-hit != 'true'
-        uses: actions/checkout@v2
-        with: 
-            path: ww3
 
       # Build WW3 spack environment
       - name: install-dependencies-with-spack
@@ -108,7 +114,7 @@ jobs:
             ~/.spack
             work_oasis3-mct
             /opt/intel
-          key: spack-${{ runner.os }}-${{ env.cache_key }}
+          key: spack-${{ runner.os }}-${{ env.cache_key }}-${{ hashFiles('ww3/model/ci/spack.yaml') }}
 
       - name: build-ww3
         run: |

